### PR TITLE
Calibration mode switch

### PR DESCRIPTION
**Pull request by mortenjc**
Calibration mode switch implemented.

This will now happen in input thread, common to all instruments.

### CODE REVIEW

File: src/common/detector/Detector.h [hunk 1]
```diff
@@ -166,4 +176,8 @@ class Detector {
   std::atomic_bool runThreads{true};
   uint32_t RuntimeStatusMask{0};
   bool CalibrationMode{false};
+
+  // Common to all EFUs
+  EV44Serializer *Serializer;
```
Comment (tbukovics) [inline src/common/detector/Detector.h:181]: So have an EV44 Serializer is not common to all EFU's. The CBM has a vector of EV44 and DA00 for example. However I planning to create common interface btw. the two and I can imagine in the future to have one common vector of serializer's.

Also this should be not in public but protected or private. It better to have a controlled function to get this data then a read/write pointer.

File: src/common/detector/Detector.h [hunk 2]
```diff
@@ -166,4 +176,8 @@ class Detector {
...
+  AR51Serializer *MonitorSerializer;
```
Comment (tbukovics) [inline src/common/detector/Detector.h:182]: I would put this as private and provide a const pointer to the users through a getter function, not a direct read/write access.

**Review by tbukovics — CHANGES_REQUESTED** [review]
I have some arguments how to store serializers in detector.h. Generally I do not like this change at the moment since some EFU's not have a basic EV44. Also how to access serializer object should be not public.

I also found in TREX this old monitoring debug mode, which is maybe outdated. There is a whole monitoring folder under the commons which should be reviewed, maybe something here is outdated after this new solution with AR51.

```c++
      if (!TREX.ADCHist.isEmpty()) {
        XTRACE(PROCESS, DEB, "Sending ADC histogram for %zu readouts",
               TREX.ADCHist.hitCount());
        ADCHistSerializer.produce(TREX.ADCHist);
        TREX.ADCHist.clear();
      }
````

File: src/common/detector/Detector.h (hunk 2 above)
Comment (mortenjc) [reply src/common/detector/Detector.h:182]: Well,there are arguments for and against. I tend to be liberal on this matter ;-)

**Comment by mortenjc** [comment]
We can discuss your comments and I can certainly see some of your arguments.

But I have very little time to spare, so I am making the improvments as practical as possible. We need to get some features in place for cold commissioning.

Generally I would like to do a major restructuring at some time in the future. You have already pointed out some issues with data sharing between *Base.cpp and  *Instrument.cpp and we also have a stat counter system that could need improvements. Also the memory pool implementation seem to prevent parallelisation.

And the old monitor code you found (in trex) should be deleted at some point. This is old stuff we don't need 
anymore.

So there is plenty of potential improvements to this eight year old system.

I would propose we collect some JIRA issues for an architecture overhaul at some point.

**Pull request merged**